lib/services/semantic-form-management.js

```diff
@@ -73,119 +73,115 @@ export class SemanticFormManagement {
-        console.log(e);
-      }
     }
 
-    if (form.status === SEMANTIC_FORM_STATUS.NEW) {
-      const ttlSource = mostRecentSource(form.sources.filter(source => source.physicalURI.endsWith('form.ttl')));
-      const configuration = (await this.configuration.sources.getLatest(ttlSource.uri)).configuration;
-
-      if (configuration && configuration.tailored.source) {
-        // Clean previously extracted data
-        const removes = await new SourceDataExtractor().extract(form.uri, configuration.specification.definition);
-        if (removes) {
-          try {
-            await this.mutator.delete(removes, form.graph);
-          } catch (e) {
-            console.warn(`Failed removing extracted source data ${uuid}`);
-            console.log(e);
-          }
-        }
-
-        // Extract new data
-        const extractors = configuration.tailored.source.content;
-        const {additions} = await new TailoredMetaDataExtractor(form).execute(extractors);
-        if (additions) {
-          try {
-            await this.mutator.insert(additions, form.graph);
-          } catch (e) {
-            console.warn(`Failed saving extracted start (source) data ${uuid}`);
-            console.log(e);
-          }
-        }
-
-        // SAVE
+    /**
+     * NOTE: generate and save tailored source TTL if is has been configured.
+     */
+    if (configuration && configuration.tailored.source) {
+      const extractors = configuration.tailored.source.content;
+      const {additions} = await new TailoredMetaDataExtractor(form).execute(extractors);
+      if (additions) {
         try {
-          await this.updateSemanticForm(uuid, form);
+          await this.mutator.insert(additions, form.graph);
         } catch (e) {
-          console.warn(`Failed updating sources for semantic-form with ${uuid}`);
+          console.warn(`Failed saving extracted start (source) data for ${uuid}`);
           console.log(e);
         }
       }
     }
 
-    const bundle = new SemanticFormBundle(form.sources);
+    // SAVE
+    try {
+      await this.updateSemanticForm(uuid, form);
+    } catch (e) {
+      console.warn(`Failed updating sources for semantic-form with ${uuid}`);
+      console.log(e);
+    }
+  }
+
+  /**
+   * Returns the {@link SemanticFormBundle} for the given UUID.
+   *
+   * TODO:  add some post-processing on the meta-data to enforce user-specific business logic.
+   *
+   * [CONSIDERED AUTH. SAFE]
+   *
+   * @param uuid
+   */
+  async getSemanticFormBundle(uuid) {
+    const form = await this.getSemanticForm(uuid);
+
+    /**
+     * NOTE: we enrich the form when we assume it to be pristine:
+     * - the status is NEW
+     * - only one source is set, we assume it to be the form.ttl (new semantic-form)
+     *
+     * TODO: make this a little more type safe / solid
+     */
+    if (form.status === SEMANTIC_FORM_STATUS.NEW || form.sources.length === 1)
```

Comment:
Do you have an idea of how that would work when we have multiple sources ? Like a v0, v1 and v2 of a form, how do we process it ?

Reply:
Is it like : on creation we take the most recent source and attach it to the form, and then here we just get it from the DB and assume there should only be one source ?